src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -4022,12 +4022,18 @@ int  java_lang_ClassLoader::unnamedModule_offset = -1;
 
 ClassLoaderData* java_lang_ClassLoader::loader_data(oop loader) {
```

Comment:
I understand the desire to minimize changes to related files (specifically the JFR files that currently include calls to java_lang_ClassLoader::loader_data() and java_lang_ClassLoader::set_loader_data(). However, I think it risks misleading future maintainers to leave these methods with their existing names when they are actually being modified to employ acquire/release semantics.

I think it would be much better to rename them consistently with the upstream patch so they are called java_lang_ClassLoader::loader_data_acquire() and java_lang_ClassLoader::release_set_loader_data() and then work those name changes through to all client calls. That not only retains consistency with upstream it also clarifies the local use.

Reply:
@adinn Thanks for reviewing. 

Make sense, updated accordingly.